c++/src/capnp/arena.h

```diff
@@ -102,15 +102,15 @@ class ReadLimiter {
   KJ_DISALLOW_COPY(ReadLimiter);
 
   KJ_ALWAYS_INLINE(void setLimit(uint64_t newLimit)) {
-#if defined(__GNUC__) || defined(__clang__)
+#if (defined(__GNUC__) || defined(__clang__)) && !defined(KJ_NO_THREADS)
```

Comment:
Are the `__atomic_*` builtins not supported on embedded systems? These are compiler intrinsics which usually don't require any special instructions, they just prevent the compiler from re-ordering operations. I thought they existed on all targets...